tools/backups2datalad/adandi.py

```diff
@@ -212,11 +214,24 @@ async def await_until_valid(self, max_time: float = 120) -> None:
             f" {json.dumps(about, indent=4)}"
         )
 
-    async def apublish(self) -> RemoteDandiset:
-        return self.for_version(
-            Version.parse_obj(
-                await self.aclient.post(f"{self.version_api_path}publish/")
-            )
+    async def apublish(self, max_time: float = 120) -> RemoteDandiset:
+        draft_api_path = f"/dandisets/{self.identifier}/versions/draft/"
+        await self.aclient.post(f"{draft_api_path}publish/")
+        log.debug(
+            "Waiting for Dandiset %s to complete publication ...", self.identifier
+        )
+        start = time()
```

Comment:
we might benefit from starting to use https://pypi.org/project/polling2/ at some point